plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -82,190 +82,414 @@ function executeCLI($scriptName, &$args) {
+			$submissionFile->setSubmissionLocale($submission->getLocale());
+			$submissionFile->setGenreId($genre->getId());
+			$submissionFile->setFileStage(SUBMISSION_FILE_PROOF);
+			$submissionFile->setAssocType(ASSOC_TYPE_REPRESENTATION);
+			$submissionFile->setData('assocId', $publicationFormatId);
+			$submissionFile->setData('createdAt', Core::getCurrentDate());
+			$submissionFile->setDateModified(Core::getCurrentDate());
+			$submissionFile->setData('mimetype', $mimeType);
+			$submissionFile->setData('fileId', $fileId);
+			$submissionFile->setData('name', pathinfo($filePath, PATHINFO_FILENAME));
+
+			// Assume open access, no price.
+			$submissionFile->setDirectSalesPrice(0);
+			$submissionFile->setSalesType('openAccess');
+
+			$submissionFileDao->insertObject($submissionFile);
+
+			// Keywords
+			$keywordsList = [$data->locale => explode(';', $data->keywords)];
+			if (count($keywordsList[$data->locale]) > 0) {
+				/** @var $submissionKeywordDao SubmissionKeywordDAO */
+				$submissionKeywordDao = DAORegistry::getDAO('SubmissionKeywordDAO');
+				$submissionKeywordDao->insertKeywords($keywordsList, $publicationId);
 			}
+
+			//Subjects
+			$subjectsList = [$data->locale => explode(";", $data->subjects)];
+			if (count($subjectsList[$data->locale]) > 0) {
+				/** @var $submissionKeywordDao SubmissionKeywordDAO */
+				$submissionSubjectDao = DAORegistry::getDAO('SubmissionSubjectDAO');
+				$submissionSubjectDao->insertSubjects($subjectsList, $publicationId);
+			}
+
+			$coverImagelocale = [];
+			$coverImage = [];
+
+			if ($data->bookCoverImage) {
+				$sanitizedCoverImageName = str_replace([' ', '_', ':'], '-', mb_strtolower($data->bookCoverImage));
+				$sanitizedCoverImageName = PKPstring::regexp_replace("/[^a-z0-9\.\-]+/", '', $sanitizedCoverImageName);
+				$sanitizedCoverImageName = basename($sanitizedCoverImageName);
+
+				$coverImage['uploadName'] = uniqid() . '-' . $sanitizedCoverImageName;
+				$coverImage['altText'] = $bookCoverImageAltText ?? '';
+
+				$destFilePath = $publicFileManager->getContextFilesPath($press->getId()) . '/' . $coverImage['uploadName'];
+				file_put_contents($destFilePath, file_get_contents($srcFilePath));
+
+				$publicationService->makeThumbnail(
+					$destFilePath,
+					$publicationService->getThumbnailFileName($coverImage['uploadName']),
+					$press->getData('coverThumbnailsMaxWidth'),
+					$press->getData('coverThumbnailsMaxHeight')
+				);
+
+				$coverImagelocale[$data->locale] = $coverImage;
+
+				$publication->setData('coverImage', $coverImagelocale);
+				$publicationDao->updateObject($publication);
+			}
+
+			// Categories
+			foreach ($dbCategories as $category) {
+				$categoryDao->insertPublicationAssignment($category->getId(), $publicationId);
+			}
+
+			echo __('plugins.importexport.csv.import.submission', ['title' => $data->title]) . "\n";
+		}
+
+		$invalidRowsFileRead = new SplFileObject($csvForInvalidRowsName, 'r');
```

Comment:
You're reading from the file which you're writing, but perhaps its content is still in memory (not flushed), to "close" a `SplFileObject`, you need to assign `null` to the variable.

But... Instead of opening the file with the errors, you can just keep a counter somewhere (e.g. `$failedRows = $processedRows - $importedRows`), which can be used on the end as a summary "1000 submissions were imported, 3 failed, see the file blabla"